lib/stripe/core_resources/payment_intent.ex

```diff
@@ -0,0 +1,290 @@
+defmodule Stripe.PaymentIntent do
+  @moduledoc """
+  Work with [Stripe `payment_intent` objects](https://stripe.com/docs/api/payment_intents).
+   You can:
+  - [Create a payment_intent](https://stripe.com/docs/api/payment_intents/create)
+  - [Retrieve a payment_intent](https://stripe.com/docs/api/payment_intents/retrieve)
+  - [Update a payment_intent](https://stripe.com/docs/api/payment_intents/update)
+  - [Confirm a payment_intent](https://stripe.com/docs/api/payment_intents/confirm)
+  - [Capture a payment_intent](https://stripe.com/docs/api/payment_intents/capture)
+  - [Cancel a payment_intent](https://stripe.com/docs/api/payment_intents/cancel)
+  - [List all payment_intent](https://stripe.com/docs/api/payment_intents/list)
+  """
+
+  use Stripe.Entity
+  import Stripe.Request
+  require Stripe.Util
+
+  @type last_payment_error :: %{
+          type: String.t(),
+          charge: String.t(),
+          code: String.t(),
+          decline_code: String.t(),
+          doc_url: String.t(),
+          message: String.t(),
+          param: String.t(),
+          source: Stripe.Card.t() | map
+        }
+
+  @type next_action :: %{
+          redirect_to_url: redirect_to_url | nil,
+          type: String.t(),
+          use_stripe_sdk: map | nil
+        }
+
+  @type redirect_to_url :: %{
+          return_url: String.t(),
+          url: String.t()
+        }
+
+  @type t :: %__MODULE__{
+          id: Stripe.id(),
+          object: String.t(),
+          amount: non_neg_integer,
+          amount_capturable: non_neg_integer,
+          amount_received: non_neg_integer,
+          application: Stripe.id() | nil,
+          application_fee_amount: non_neg_integer | nil,
+          canceled_at: Stripe.timestamp() | nil,
+          cancellation_reason: String.t() | nil,
+          capture_method: String.t(),
+          charges: Stripe.List.t(Stripe.Charge.t()),
+          client_secret: String.t(),
+          confirmation_method: String.t(),
+          created: Stripe.timestamp(),
+          currency: String.t(),
+          customer: Stripe.id() | Stripe.Customer.t() | nil,
+          description: String.t() | nil,
+          last_payment_error: last_payment_error | nil,
+          livemode: boolean,
+          metadata: Stripe.Types.metadata(),
+          next_action: next_action | nil,
+          on_behalf_of: Stripe.id() | Stripe.Account.t() | nil,
+          payment_method_types: list(String.t()),
+          receipt_email: String.t() | nil,
+          review: Stripe.id() | Stripe.Review.t() | nil,
+          shipping: Stripe.Types.shipping() | nil,
+          source: Stripe.Card.t() | map,
+          statement_descriptor: String.t() | nil,
+          status: String.t(),
+          transfer: Stripe.id() | Stripe.Transfer.t() | nil,
```

Comment:
It looks like this should be `:transfer_data` https://stripe.com/docs/api/payment_intents/object#payment_intent_object-transfer_data

Reply:
thanks for that. i will change it soon!

Reply:
just had a look into it again and I took this example from the charge object. [charge](https://github.com/code-corps/stripity_stripe/blob/master/lib/stripe/core_resources/charge.ex#L91)

[stripe reference](https://stripe.com/docs/api/charges/object#charge_object-transfer_data)

Reply:
Looks like charge has `transfer` && `transfer_data` (to be added).  So in this case, payment intent just has `transfer_data` . 👍

Reply:
i see, going to change it very soon 👍

Reply:
@snewcomer i've added transfer_data to charge and payment_intent